main.bal

```diff
@@ -377,13 +379,13 @@ service / on new http:Listener(serverPort) {
         database:ExperimentFull[] experiments;
 
         transaction {
-            experimentCount = check database:getExperimentCount();
+            experimentCount = check database:getExperimentCount(name = name);
```

Comment:
a better internal name for the variable/parameter `name` would be `nameFilter` or make it more consistent overall and name it `search`